### DISCUSSION TITLE
Ant combat

### DISCUSSION THREAD

**Discussion by TimJentzsch** [Leafwing-Studios/Emergence#169] [Ideas]
[Combat is a key aspect where our game could shine among the competition](https://leafwing-studios.github.io/Emergence/high-level/genre-analysis.html#common-problems--room-for-improvement).
It should be well integrated and not only feel like a frustrating expansion gate.

Here's one idea for a more interesting combat system: Ant combat.
In this discussion I will assume that ants are the "main units" of the game, but the concept could be transferred to other organisms.

The key concept here is that we are not the only ant colony in the area, so we'll have to compete for the available natural resources.
Each colony has a _territory_ which they are going to defend.
If we don't expand into their territory, we will see little to no attacks on our colony, but this can severely restrict the resources available to us.
If we use those resources efficiently, we might get far without any combat though.

If we want to expand, we'll have to fight for it. The closer we get to the center of an enemy colony, the harder they will try to defend the area.
Our goal will be to defeat the enemy by killing their _ant queen(s)_.
Once we accomplish that, we can freely use the territory of the other colony.
Importantly, we can also reuse some structures they have built, so we get an immediate _reward_ for our accomplishment (instead of only having more space like in Factorio).

There will be multiple sizes of enemy colonies.
Larger colonies will be much harder to defeat, but also provide much more resources, structures, and space to capture.
One strategy will be to stay away from the large colonies in the early game and expand into other directions.
Once the hive has grown stronger, more and more enemies will become valid targets for attack.

A system like this would integrate quite naturally into the theme of the game.
It also mostly allows players to play at their own pace, a slow player can take more time to grow stronger before mounting attacks.
We might also provide the opportunity to choose between the attack or assimilation of a different species.
Assimilation could be the "peaceful" way that takes a lot longer, combat is faster but also risky and we destroy some things that we could otherwise use.

Whether (or how often) other colonies also mount attacks at us outside of their territory could be a setting for the player to configure.

**Comment by alice-i-cecile** [comment]
This is well-reasoned, and I think presents a compelling possible direction! We need to figure out performance constraints before settling on this: I think that combat and especially alternate colonies would be one of the first things to cut if we're tight. 

I also think predator attacks are another good avenue for well-integrated combat.
